config-builder/src/lib.rs

```diff
@@ -0,0 +1,248 @@
+    derive_config_builder_impl(input).unwrap_or_else(|err| err.to_compile_error().into())
+}
+
+fn derive_config_builder_impl(input: TokenStream) -> syn::Result<TokenStream> {
+    let input = {
+        let ts: proc_macro2::TokenStream = proc_macro2::TokenStream::from(input);
+        syn::parse2::<DeriveInput>(ts)?
+    };
+    let struct_ident = input.ident.clone();
+    let builder_ident = format_ident!("{}Builder", struct_ident);
+
+    // Extract struct fields
+    let fields = match input.data {
+        Data::Struct(data_struct) => match data_struct.fields {
+            Fields::Named(fields_named) => fields_named.named,
+            _ => {
+                return Err(Error::new_spanned(
+                    struct_ident,
+                    "ConfigBuilder only supports structs with named fields",
+                ));
+            }
+        },
+        _ => {
+            return Err(Error::new_spanned(
+                struct_ident,
+                "ConfigBuilder only supports structs",
+            ));
+        }
+    };
+
+    // Builder fields - all wrapped in Option
+    let builder_fields = fields.iter().map(|f| {
+        let name = &f.ident;
+        let ty = &f.ty;
+        quote! {
+            #name: Option<#ty>
+        }
+    });
+
+    let setters = fields.iter().map(|f| {
+        let name = &f.ident;
+        let ty = &f.ty;
+
+        if is_string_type(ty) {
+            quote! {
+                /// Sets the `#name` field.
+                pub fn #name(mut self, #name: impl Into<String>) -> Self {
+                    self.#name = Some(#name.into());
+                    self
+                }
+            }
+        } else {
+            quote! {
+                /// Sets the `#name` field.
+                pub fn #name(mut self, #name: #ty) -> Self {
+                    self.#name = Some(#name);
+                    self
+                }
+            }
+        }
+    });
+
+    // Build method -> merge with defaults
+    let build_fields = fields.iter().map(|f| {
+        let name = &f.ident;
+        quote! {
+            #name: self.#name.unwrap_or(default.#name)
+        }
+    });
+
+    let getters = fields.iter().map(|f| {
+        let name = &f.ident;
+        let ty = &f.ty;
+
+        if is_string_type(ty) {
+            quote! {
+                /// Returns the value of the `#name` field as a string slice.
+                pub fn #name(&self) -> &str {
+                    &self.#name
+                }
+            }
+        } else if is_copy_type(ty) {
+            quote! {
+                /// Returns the value of the `#name` field.
+                pub fn #name(&self) -> #ty {
+                    self.#name
+                }
+            }
+        } else {
+            quote! {
+                /// Returns a reference to the `#name` field.
+                pub fn #name(&self) -> &#ty {
+                    &self.#name
+                }
+            }
+        }
+    });
```

Comment:
```suggestion
        if is_copy_type(ty) {
            quote! {
                /// Returns the value of the `#name` field.
                pub fn #name(&self) -> #ty {
                    self.#name
                }
            }
        } else {
            quote! {
                /// Returns a reference to the `#name` field.
                pub fn #name(&self) -> &#ty {
                    &self.#name
                }
            }
        }
```

Why would we need the String value as a slice? We can return the String itself instead of making extra conditions. It will return as a reference anyway. You don't need this extra condition